axiom/resource_notifier_test.go

```diff
@@ -0,0 +1,128 @@
+package axiom
+
+import (
+	"os"
+	"testing"
+
+	ax "github.com/axiomhq/axiom-go/axiom"
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNotifiers(t *testing.T) {
+	client, err := ax.NewClient()
+	assert.NoError(t, err)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"axiom": providerserver.NewProtocol6WithError(NewAxiomProvider()),
+		},
+		CheckDestroy: testAccCheckAxiomResourcesDestroyed(client),
+		Steps: []resource.TestStep{
+			{
+				Config: testConfigNotifier("slack", ` {
+					slack = {
+						slack_url = "https://hooks.slack.com/services/T00000000/B00000000/XXXXXXXXXXXXXXXXXXXXXXXX"
+					} }`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourcesCreatesCorrectValues(client, "axiom_notifier.slack", "properties.slack.slack_url", "properties.slack.slackUrl"),
+				),
+			},
+			{
+				Config: testConfigNotifier("discord", ` {
+					discord = {
+						discord_channel = "general"
+						discord_token = "fake_discord_token"
+					} }`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourcesCreatesCorrectValues(client, "axiom_notifier.discord", "properties.discord.discord_channel", "properties.discord.discordChannel"),
+					testAccCheckResourcesCreatesCorrectValues(client, "axiom_notifier.discord", "properties.discord.discord_token", "properties.discord.discordToken"),
+				),
+			},
+			{
+				Config: testConfigNotifier("discord_webhook", ` {
+					discord_webhook = {
+						discord_webhook_url = "https://discord.com/api/webhooks/1234567890/abcdefghijklmnopqrstuvwxyz"
+					} }`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourcesCreatesCorrectValues(client, "axiom_notifier.discord_webhook", "properties.discord_webhook.discord_webhook_url", "properties.discordWebhook.discordWebhookUrl"),
+				),
+			},
+			{
+				Config: testConfigNotifier("email", ` {
+					email = {
+						emails = ["test@example.com"]
+					} }`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourcesCreatesCorrectValues(client, "axiom_notifier.email", "properties.email.emails.0", "properties.email.emails.0"),
+				),
+			},
+			{
+				Config: testConfigNotifier("opsgenie", ` {
+					opsgenie = {
+						api_key = "fake_opsgenie_api_key"
+						is_eu = false
+					} }`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourcesCreatesCorrectValues(client, "axiom_notifier.opsgenie", "properties.opsgenie.api_key", "properties.opsgenie.apiKey"),
+
+					// is EU is not in the response?
```

Comment:
@toppercodes is this expected?

Reply:
It's the ommit empty on the swagger again
False is the default value so is omitted

Reply:
have updated the test